### PR TITLE
Here Comes The Money (Baby's first baycode port)

### PR DESCRIPTION
**Pull request by Sun-Soaked**
## About The Pull Request

- Replaces our stacks of individual denominations of Space Cash with a single reconfigurable object that can be freely split or merged to any value
These new space cash piles have a variable icon that changes to include the exact denomination of cash held within

- Ports messier, richer cash sprites from CEV Eris. They're pretty good but the coins may end up having to be fixed up a bit.
image

- Replaces some holochip interactions with Space Cash

- Adds sound effects to picking up and dropping piles of money (different for bills and coins!)

- Some cash drops have become randomized. This serves no practical purpose (besides a mild nerf to looted money) but looks more visually interesting then flat 000 values with the new texture system

## Why It's Good For The Game

Immersive coin counting action
Let me know if you think we should deinflate the economy ingame so people end up using the coins more

## Changelog

:cl:
add: Physical money has been reworked one central item, instead of several smaller denominations. This item has a variable icon based on the amount of money contained within.
tweak: The central galactic bank has declared that material coins no longer have a monetary value. You can no longer print money with a coin press, if you somehow find one.
tweak: Many money drops are now slightly randomized.
tweak: Some interactions that used holocurrency now use hard cash instead.
soundadd: Adds new sound effects for dropping and picking up bills and coins.
imageadd: Adds new physical cash sprites.
code: Cash.dm has been moved up a directory out of the Stacks dir, on account of no longer being a stack subtype
/:cl:

### CODE REVIEW

**Comment by Sun-Soaked** [comment]
....I forgot to fix all the maps. Woe upon me

**Comment by axelzonvolt** [comment]
next PR should change the vending machines so that instead of drawing from the account of the spaceman inquestion you have to either swipe your ID or manually input space coins to receive your space twinkies.

**Comment by thgvr** [comment]
this is awesome

File: code/game/machinery/slotmachine.dm [hunk 1]
```diff
@@ -254,14 +254,15 @@
 /obj/machinery/computer/slot_machine/proc/give_prizes(usrname, mob/user)
 	var/linelength = get_lines()
 
-	if(reels[1][2] + reels[2][2] + reels[3][2] + reels[4][2] + reels[5][2] == "[SEVEN][SEVEN][SEVEN][SEVEN][SEVEN]")
+	if(jackpots >= 0)
+	//if(reels[1][2] + reels[2][2] + reels[3][2] + reels[4][2] + reels[5][2] == "[SEVEN][SEVEN][SEVEN][SEVEN][SEVEN]")
```
Comment (MarkSuckerberg) [inline code/game/machinery/slotmachine.dm]: What's this changed for?

File: code/game/objects/items/cash.dm [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+			drop_sound = 'sound/items/handling/dosh_drop.ogg'
+			pickup_sound =  'sound/items/handling/dosh_pickup.ogg'
+		else
+			name = "[value] credits"
+			gender = NEUTER
+			desc = "An executive's ransom made manifest."
+			drop_sound = 'sound/items/handling/dosh_drop.ogg'
+			pickup_sound =  'sound/items/handling/dosh_pickup.ogg'
+
+/obj/item/spacecash/bundle/attack_self()
```
Comment (MarkSuckerberg) [inline code/game/objects/items/cash.dm:113]: Can we get an altclick to do this same behaviour? It'd certainly be nice to have to draw from briefcases, piles on the ground, etc.

File: code/game/objects/items/storage/boxes.dm [hunk 3]
```diff
@@ -1427,7 +1427,7 @@
 		/obj/item/construction/rcd/combat/admin=1,\
 		/obj/item/pipe_dispenser=1,\
 		/obj/item/card/emag=1,\
-		/obj/item/stack/spacecash/c1000=50,\
+		/obj/item/spacecash/bundle/mediumrand=50,\
```
Comment (MarkSuckerberg) [inline code/game/objects/items/storage/boxes.dm]:
```suggestion
		/obj/item/spacecash/bundle/c10000=5,\
```

File: code/game/objects/items/storage/wallets.dm [hunk 4]
```diff
@@ -120,5 +120,5 @@
 	icon_state = "random_wallet"
 
 /obj/item/storage/wallet/random/PopulateContents()
-	new /obj/item/holochip(src, rand(5,30))
+	new /obj/item/spacecash/bundle(src, rand(5,30))
```
Comment (MarkSuckerberg) [inline code/game/objects/items/storage/wallets.dm]:
```suggestion
	new /obj/item/spacecash/bundle/pocketchange(src)
``` 
fear my wallet buff for the sake of standardisation

**Review by MarkSuckerberg — CHANGES_REQUESTED** [review]
Just a few more things

**ALSO** the icon state for the 50 credit bill seems to have a stray apostrophe at the end, so you might wanna check that

File: code/game/machinery/slotmachine.dm (hunk 1 above)
Comment (Sun-Soaked) [reply code/game/machinery/slotmachine.dm]: oh, I was testing something, sorry. Lemme fix that

Comment (Sun-Soaked) [reply code/game/machinery/slotmachine.dm]: fixed in my next pull

File: code/game/objects/items/cash.dm (hunk 2 above)
Comment (Sun-Soaked) [reply code/game/objects/items/cash.dm:113]: an almost utter copy-paste into /altclick() has achieved this functionality perfectly
pushing now

**Review by MarkSuckerberg — APPROVED** [review]

**Pull request merged**